File: web/src/gql/fragments/asset.ts
import { gql } from "@apollo/client";

import { assetFileFragment, integrationFragment, threadFragment } from "@reearth-cms/gql/fragments";

export const assetFragment = gql`
  fragment assetFragment on Asset {
    id
    fileName
    projectId
    createdAt
    createdBy {
      ... on User {
        id
        name
        email
      }
      ... on Integration {
        ...integrationFragment
      }
    }
    size
    previewType
    uuid
    url
    thread {
      ...threadFragment
    }
    archiveExtractionStatus
    public
  }
  ${assetFileFragment}
  ${integrationFragment}
  ${threadFragment}
`;

export default assetFragment;

File: server/internal/adapter/internalapi/internalapimodel/asset.go
package internalapimodel

import (
	pb "github.com/reearth/reearth-cms/server/internal/adapter/internalapi/schemas/internalapi/v1"
	"github.com/reearth/reearth-cms/server/pkg/asset"
	"google.golang.org/protobuf/types/known/timestamppb"
)

func ToAsset(a *asset.Asset) *pb.Asset {
	if a == nil {
		return nil
	}
	return &pb.Asset{
		Id:                      a.ID().String(),
		Uuid:                    a.UUID(),
		ProjectId:               a.Project().String(),
		Size:                    a.Size(),
		Filename:                a.FileName(),
		PreviewType:             a.PreviewType().StringRef(),
		Url:                     a.AccessInfo().Url,
		ArchiveExtractionStatus: a.ArchiveExtractionStatus().StringRef(),
		Public:                  a.AccessInfo().Public,
		CreatedAt:               timestamppb.New(a.CreatedAt()),
	}
}
